### ISSUE TITLE
Support image relocation 

### ISSUE THREAD

**Issue by jeremyrickard** [deislabs/porter#543]
Images referenced by a CNAB bundle MAY be [relocated](https://github.com/deislabs/cnab-spec/blob/master/103-bundle-runtime.md#image-relocation) , for example by copying them to a private registry. This process results in a mapping file:

```
{  "gabrtv/microservice@sha256:cca460afa270d4c527981ef9ca4989346c56cf9b20217dcea37df1ece8120687": "my.registry/microservice@sha256:cca460afa270d4c527981ef9ca4989346c56cf9b20217dcea37df1ece8120687",
  "technosophos/helloworld:0.1.0": "my.registry/helloworld:0.1.0"
}
```

That mapping file _should_ be mounted into an invocation image at: `

The spec further says: 

```
The run tool MAY use this file to modify its behavior. For example, a run tool MAY substitute image references using the mapping in this file.
```

We should enhance porter to enable a relocation file to be provided and mounted into an invocation image,  then also modify the porter runtime so that anything in the relocation file can be used within the execution of a bundle operation. Ideally, if we implement #501, people that build bundles that reference images can simply refer to them like ``{{ bundle.images.SOMENAME}}``, like in a call to the `helm` mixin or some other mixin that needs to use images, and then Porter will helpfully provide them with the correct reference.

**Comment by vdice** [comment]
@carolynvs, @jeremyrickard and I discussed putting this back in the Backlog for now.  As of writing, our main needs lie in bundle archiving and bundle publishing (from an archive) which _may_ publish bundle images to a new registry but currently can do so with updated digests/shas.  Therefore, as this ticket (and image relocation in general) brings in a more complex approach to republishing (most pertinently, preserving digests which will go in tandem with forthcoming security guarantees), we've decided to delay this work.
